src/Database/IdentifierInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.4.1
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database;
+
+/**
+ * Represents a single identifier name in the database
+ */
+interface IdentifierInterface extends ExpressionInterface
```

Comment:
I'd say in that case it would then be more of a "has to", rather than an "allow to", given that the query builder/compiler expects identifiers to implement `ExpressionInterface` in order to be compilable.

Without an additional interface that does extend `ExpressionInterface`, or a public identifier expression implementation, the problem of the `identifier()` API being insufficiently typed would remain.